src/port1.0/portconfigure.tcl

```diff
@@ -1372,6 +1372,9 @@ proc portconfigure::configure_main {args} {
         global configure.${flags} configure.universal_${flags}
     }
 
+    # Hack to export DEVELOPER_DIR for all child processes
+    set ::env(DEVELOPER_DIR) [option configure.developer_dir]
```

Comment:
We might not always run the configure phase (for example if it has been completed in a previous invocation of `port`), so just setting this in configure and assuming it will stay set for all following phases is probably wrong.

Reply:
Putting it in main works and I could install libnetpbm in tracemode, but I had to hardcode the path. I can't access the configure.developer_dir variable. Each case, it produces a warning/error.

- hardcode produces warning at runtime in macports::setxcodeinfo
```
set ::env(DEVELOPER_DIR) "/Library/Developer/CommandLineTools/"
...
Warning: xcodebuild exists but failed to execute
```
So maybe it is overkill to put it in main, since we ultimately just want to affect the system -W calls. Which part of the codebase invokes the e.g post-build {} blocks of a Portfile?

- configure.developer_dir errors at runtime
```
...
set ::env(DEVELOPER_DIR) [option configure.developer_dir]
...
Error: Unable to open port: can't read "configure.developer_dir": no such variable
```
The solution would be to option configure.developer_dir and default configure.developer_dir before this code on main, but the naming wouldn't make sense anymore. Or maybe explore modifying system.c instead. What do you think?